lib/iris/tests/unit/plot/test__check_bounds_contiguity_and_mask.py

```diff
@@ -88,20 +90,20 @@ def test_2d_contiguous_atol(self):
         discontiguity_check.assert_called_with(atol=1e-3)
 
     def test_2d_discontigous_masked(self):
-        # Test a 2D coordinate which is discontiguous but masked at
-        # discontiguities.
+        # Test that a 2D coordinate which is discontiguous but masked at
+        # discontiguities doesn't error.
         cube = sample_2d_latlons()
         make_bounds_discontiguous_at_point(cube, 3, 4)
-        iplt._check_bounds_contiguity_and_mask(cube.coord('longitude'),
-                                               cube.data)
+        self.assertIsNone(iplt._check_bounds_contiguity_and_mask(
+            cube.coord('longitude'), cube.data))
 
     def test_2d_discontigous_unmasked(self):
         # Test a 2D coordinate which is discontiguous and unmasked at
         # discontiguities.
         cube = sample_2d_latlons()
         make_bounds_discontiguous_at_point(cube, 3, 4)
         msg = 'coordinate are not contiguous'
-        cube.data[3, 4] = ma.nomask
+        cube.data.mask[3, 4] = ma.nomask
```

Comment:
Yeah I was a little unsure about the specifics of all this.  It's definitely a masked array at line 106ish because `make_bounds_discontiguous_at_point` masks it, but there are some secret and confusing corners of masked array operations.

Anyway, I've reverted it to `cube.data[3, 4] = ma.nomask` now.  Thanks for the advice.